types/ot/index.d.ts

```diff
@@ -6,36 +6,47 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 export class Selection {
-    createCursor(position: number): Selection;
+    ranges: Selection.Range[];
+
+    equals(other: Selection): boolean;
+
+    static createCursor(position: number): Selection;
 }
 
 export namespace Selection {
     class Range {
-        constructor(anchor: number, head: number);
-        transform(operation: TextOperation): Range;
         anchor: number;
         head: number;
+
+        constructor(anchor: number, head: number);
+
+        equals(other: Range): boolean;
+        isEmpty(): boolean;
+        transform(operation: TextOperation): Range;
+
+        static fromJSON(object: { anchor: number; head: number; }): Range;
     }
 }
 
 export type SerializedTextOperation = Array<string | number>;
 
 export class TextOperation {
-    delete(length: number): TextOperation;
-    insert(str: string): TextOperation;
-    retain(length: number): TextOperation;
-
     baseLength: number;
+    ops: SerializedTextOperation;
     targetLength: number;
 
     apply(code: string): string;
     compose(operation: TextOperation): TextOperation;
-
-    static transform(left: TextOperation, right: TextOperation): TextOperation;
-    static isRetain(operation: TextOperation): boolean;
-    static isInsert(operation: TextOperation): boolean;
-    static isDelete(operation: TextOperation): boolean;
+    delete(length: number): TextOperation;
+    insert(str: string): TextOperation;
+    retain(length: number): TextOperation;
+    toJSON(): SerializedTextOperation;
 
     static fromJSON(operation: SerializedTextOperation): TextOperation;
-    toJSON(): SerializedTextOperation;
+    static isDelete(operation: string | number): boolean;
+    static isInsert(operation: string | number): boolean;
+    static isRetain(operation: string | number): boolean;
+    static transform(left: TextOperation, right: TextOperation): TextOperation;
 }
+
+export const version: '0.0.15';
```

Comment:
Would `string` be more appropriate?  Doing it this way would seem to require that every version have a distinct .d.ts file, even if this is the only change.  And callers are probably consuming it as a string?

Reply:
@amcasy `ot` is last updated about 6 years ago, so I don't think it matters here.
It's an extra confirmation to know these types are about the latest version.

Reply:
On the flip side, what's the benefit of giving the constant a literal type?  Maybe it's to cause some old code to be flagged as unreachable?
```ts
if (version === "0.0.14") {
    // unreachable
}
```

Reply:
Good point, I'll change it to just string 🙂